script.py

```
@@ -1,553 +1,1021 @@
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    def _run_cmd(
+        self,
+        cmd: list[str],
+        *,
+        cwd: Optional[str] = None,
+        stdin_data: Optional[str] = None,
+    ) -> subprocess.CompletedProcess:
+        """Shared helper to run shell commands safely."""
+        try:
+            return subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)